src/python/grpcio/grpc/_server.py

```diff
@@ -1061,15 +1060,14 @@ def _handle_with_method_handler(
                 return _handle_stream_unary(
                     rpc_event, state, method_handler, thread_pool
                 )
+        elif method_handler.response_streaming:
+            return _handle_unary_stream(
+                rpc_event, state, method_handler, thread_pool
+            )
         else:
-            if method_handler.response_streaming:
-                return _handle_unary_stream(
-                    rpc_event, state, method_handler, thread_pool
-                )
-            else:
-                return _handle_unary_unary(
-                    rpc_event, state, method_handler, thread_pool
-                )
+            return _handle_unary_unary(
```

Comment:
I'd get rid of this last else statement too and unindent this return, but GitHub suggestions won't work with mixed diff here.